proposals/0357-alpenglow_validator_admission_ticket.md

```diff
@@ -0,0 +1,219 @@
+similar to pre-Alpenglow consensus.
+
+## New Terminology
+
+- **Validator Admission Ticket(VAT)**: The 1.6 SOL charged once per epoch to
+every validator eligible to participate in the next epoch
+
+## Different Vote Related Accounts
+
+Before Alpenglow, we have the following accounts related to voting (the
+identity account and vote authority account can be the same):
+
+- Vote account for saving all the vote states and receiving commission
+
+- Identity account for receiving block rewards
+
+- Vote authority account for signing vote transactions
+
+After Alpenglow, we will still have roughly the same accounts serving the same
+purpose:
+
+- Vote account: This account must contain the correct BLS public key
+corresponding to vote authority keypair. It continues to keep all the vote
+credits and all validator identity/authority/commission information updates
+happen here, but it doesn’t contain vote information any more.
+
+- Identity account: This is the account for the 1.6 SOL VAT; it continues
+receiving block rewards and commissions.
+
+- Vote authority account: For signing BLS messages
+
+## Detailed Design
+
+### What is considered a valid vote account
+
+The VAT discussion in SIMD 326 proposes:
+
+- The VAT is a temporary measure before we implement a better mechanism
+
+- There is a strict limit of the 2,000 highest staked validators
+
+- Before being admitted to an epoch, the 1.6 SOL per epoch VAT fee is deducted
+  from each validator and burned
+
+Another requirement (not explicitly mentioned in SIMD 326 but in the Alpenglow
+White Paper) is that Alpenglow uses BLS to aggregate signatures. So a future
+Alpenglow vote account must contain a valid BLS public key deduced from its
+current vote authority ed25519 private key.
+
+In summary, a valid vote account in an Alpenglow epoch must contain:
+
+- a BLS public key
+
+- at least 1.6 SOL VAT fee plus the necessary storage rent amount for a new
+epoch in its corresponding identity account
+
+When the staked validators for a new epoch are calculated, the leader will
+perform the following operations:
+
+- When there are more than 2,000 valid validators, sort all valid vote accounts
+by descending order of stake. If some validator with stake S is in position
+2001, then we remove all validators with stake S and less. If there are fewer
+than 2,000 valid validators, pick all of them.
+
+- Deduct 1.6 SOL VAT fee from each picked vote identity account once
+
+- Mark the fee burned and write the result into the bank
+
+All validators must perform the same operation, or the resulting bankhash will
+be different.
+
+### How to implement the checks
+
+1. When a new bank crosses an epoch boundary (bank.epoch() >
+parent_bank.epoch()), calculate the participating staked validators for the
+next epoch (bank.epoch() + 1). This is the same as now.
+
+2. Perform stake activation and deactivation so that the intended stake values
+are used for the new epoch.
```

Comment:
the timing of this operation is insufficiently specified. an implementer needs to know when it happens in relation to at least feature gate activation, rewards payouts and stake (de)activation. not sure if there are other dependencies that could be in conflict. a lot happens on the epoch boundary